apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataReader.cs

```diff
@@ -254,6 +254,24 @@ public async Task<int> GetOrdinalAsync(string name, CancellationToken cancellati
             return _fieldIndex[name];
         }
 
+        /// <summary>
+        /// Gets the read timstamp if <see cref="TimestampBound.ReturnReadTimestamp" /> was specified.
+        /// </summary>
+        /// <remarks>
+        /// The read timestamp is only available after at least one result is
+        /// read. If the returned results are empty, then a read timestamp
+        /// won't be available.
+        /// </remarks>
+        /// <returns>The value converted to a <see cref="Timestamp"/>.</returns>
+        /// <exception cref="InvalidOperationException">A read timestamp is unavailable,
+        /// possibly becuase <see cref="TimestampBound.ReturnReadTimestamp" /> was not set.</exception>
+        public Timestamp GetReadTimestamp()
```

Comment:
Given that `PopulateMetadata()` runs `PopulateMetadataAsync()` if it's not already set, perhaps we need an async version of this call as well? We could do that later, of course.

Reply:
Thinking about this more, if the timestamp is only actually provided when you call Read or ReadAsync, maybe we shouldn't call PopulateMetadata at all, and just use the existing cached version? Under what circumstances would PopulateMetadata provide something here without it having been cached before? (I may have misunderstood something.)

Reply:
I think the reason we're calling it is because maybe the metadata is not already populated. It seems that all of the code that relies on a metadata field calls this method as a precaution just in case no one else has called it before (the value is cached after the first call so we're using a cached value if it's already been called before).

Reply:
I added an async version.